### PR TITLE
remove hardcoded stdlib packages

### PR DESCRIPTION
**Pull request by xytan0056**
currently the standard list is hardcoded, we can load this from x/tools/go/packages, which is already a dependency.
This means whenever for the same release, different toolchains install could have different the package list. So we don't have to manually maintain this list to include everything stdlib

unit tested

### CODE REVIEW

File: pkg/section/parser.go [hunk 1]
```diff
@@ -22,7 +22,7 @@ func Parse(data []string) (SectionList, error) {
 		if s == "default" {
 			list = append(list, Default{})
 		} else if s == "standard" {
-			list = append(list, Standard{})
+			list = append(list, NewStandard())
```
Comment (daixiang0) [inline pkg/section/parser.go:25]: No need to change this.

File: pkg/section/standard.go [hunk 2]
```diff
@@ -3,14 +3,30 @@ package section
 import (
 	"github.com/daixiang0/gci/pkg/parse"
 	"github.com/daixiang0/gci/pkg/specificity"
+	"golang.org/x/tools/go/packages"
 )
 
 const StandardType = "standard"
 
-type Standard struct{}
+type Standard struct {
```
Comment (daixiang0) [inline pkg/section/standard.go:11]: Make `Standard` as a type alias rather than struct.

File: pkg/section/standard_test.go [hunk 3]
```diff
@@ -8,13 +8,13 @@ import (
...
 func TestStandardPackageSpecificity(t *testing.T) {
 	testCases := []specificityTestData{
-		{"context", Standard{}, specificity.StandardMatch{}},
-		{"contexts", Standard{}, specificity.MisMatch{}},
-		{"crypto", Standard{}, specificity.StandardMatch{}},
-		{"crypto1", Standard{}, specificity.MisMatch{}},
-		{"crypto/ae", Standard{}, specificity.MisMatch{}},
-		{"crypto/aes", Standard{}, specificity.StandardMatch{}},
-		{"crypto/aes2", Standard{}, specificity.MisMatch{}},
+		{"context", NewStandard(), specificity.StandardMatch{}},
```
Comment (daixiang0) [inline pkg/section/standard_test.go:11]: Please make it as a const, no need to call each time.

**Comment by daixiang0** [comment]
I have a question: if we use 1.9 toolchain, users use 1.10, the newer packages cannot be detected, is it right? If so, we can not use this way.

**Comment by xytan0056** [comment]
> I have a question: if we use 1.9 toolchain, users use 1.10, the newer packages cannot be detected, is it right? If so, we can not use this way.

indeed. Maybe I need to reword the PR. But wouldn't we already have this issue with current implementation? The hardcoded list cannot cover all packages, so might as well make it accustomed to users' toolchains. e.g.
```
func main() {
        pkgs, err := packages.Load(nil, "std")
        if err != nil {
                panic(err)
        }

        fmt.Println(pkgs)
        // go1.16.1 run ./... => 212
        // go1.18.3 run ./... => 233
        // go1.19.4 run ./... => 241
        fmt.Println(len(pkgs)) 
}
```
I believe when users run `go1.x.x install` it'll refer to that version

**Comment by daixiang0** [comment]
You are right, please fix CI then we can get it.

**Comment by daixiang0** [comment]
@xytan0056 thanks for your contribution!

**Pull request merged**